services/sidecar/src/simcore_service_sidecar/celery_task.py

```diff
@@ -0,0 +1,90 @@
+from asyncio import CancelledError
+from pprint import pformat
+from typing import Optional
+
+from celery import Celery, states
+
+from .celery_log_setup import get_task_logger
+from .cli import run_sidecar
+from .config import CPU_QUEUE_NAME, GPU_QUEUE_NAME, MPI_QUEUE_NAME
+from .core import task_required_resources
+from .utils import wrap_async_call
+
+log = get_task_logger(__name__)
+
+
+def entrypoint(
```

Comment:
see my comment above. I think binding an ``app`` to the function could be done
a) [functools.partial](https://docs.python.org/3.6/library/functools.html#functools.partial)
b) creating your own *high-order-function*
```python
 def create_some_fun(app) -> Callable:
      def some_fun(a, b,c):
               ...
               return foo(app) - c + a -b
      return some_fun
```

Reply:
added comment